src/main/flight/mixer.c

```diff
@@ -814,6 +814,22 @@ float applyThrottleLimit(float throttle)
     return throttle;
 }
 
+void applyAirMode(float *motorMix, float motorMixMax)
+{
+    float absThrottle = ABS(throttle);
+    float normalizationFactor = motorMixRange > 1.0f && hardwareMotorType != MOTOR_BRUSHED ? motorMixRange : 1.0f;
+    float motorMixDelta = 0.5f * motorMixRange;
+    for (int i = 0; i < motorCount; ++i) {
+        motorMix[i] += motorMixDelta - motorMixMax; // let's center the values exactly around the zero
+        float result = motorMix[i] + scaleRangef(absThrottle, 0.0f, 1.0f, motorMixDelta, - motorMixDelta); // this is the actual AirMode trick
```

Comment:
I do have one question about scaleRangef, I've never used this function before. Do you happen to know the kind of cpu load that this adds vs scaling things yourself? I've never seen this function used until you started showing its potential to me, but I'm wondering if I've never encountered it because it uses more cpu then we would like. Just a thought and I figured you might know.

Reply:
Good question.
I don't know if we would obtain better performances by using a custom/in-line scaling, the compiler should optimize the formula the scalerangef uses (just a couple of additions and multiplications). Personally I prefer legibility when I'm not sure if there is actually a performance issue, but we could substitute it with a macro etc. if needed!